sentry_kube/cli/pg/create_user.py

```diff
@@ -124,35 +127,46 @@ def upload_plaintext_to_k8s_secret(
     users: dict[str, dict[str, str]],
     secret_name: str
 ) -> None:
-    # TODO: create if not exists
-    # api.create_namespaced_secret(namespace, body)
+    try:
+        secret = api.read_namespaced_secret(namespace="default", name=secret_name)
+    except ApiException as exc:
+        if exc.status == 404:
+            print(f"Secret `default/{secret_name}` does not exist, creating...")
+            body = V1Secret()
+            body.metadata = {"name": secret_name}
+            body.data = {}
+            api.create_namespaced_secret(namespace="default", body=body)
+            secret = api.read_namespaced_secret(namespace="default", name=secret_name)
+        else:
+            raise
```

Comment:
Maybe define a new exception FailedSecretCreationException and raise that instead of a generic raise. That way when you write tests its easier to assert what assertion got raised.